src/tests/api/session.test.ts

```diff
@@ -0,0 +1,67 @@
+import { assert200OK, assert400BadRequest, assert401Unauthorized } from "../../utils/apiCommonAssertions";
+import { sessionPost, stubStartPost } from "../../utils/apiTestSteps";
+
+describe("Happy Path Get SessionID", () => {
+	let stubResponse: any;
+
+	beforeEach( async () =>{
+		const requestBody = {
+			target:"https://unb2uhs7rl.execute-api.eu-west-2.amazonaws.com/dev/",
+		};
+		stubResponse = await stubStartPost(requestBody);
+
+	});
+    
+	it("Given valid endpoint has been received", async () => {
+
+		const response = await sessionPost(stubResponse.data.clientId, stubResponse.data.request);
+
+		assert200OK({ status: response.status, statusText: response.statusText });
+	});
+
+});
+
+
+describe("UnHappy Path", () => {
+
+	it("401 response - Empty Request in request body", async () => {
+
+		const requestBody = {
+			target:"https://unb2uhs7rl.execute-api.eu-west-2.amazonaws.com/dev/",
+		};
+		const stubResponse = await stubStartPost(requestBody);
+
+		const response = await sessionPost(stubResponse.data.clientId, "");
+
+		assert401Unauthorized(response.status, response.statusText);
+
+		expect(response.data.message).toBe("Invalid request: Request failed to be decrypted");
```

Comment:
If the rest body is empty the oauth spec says we should return the user back with the error `invalid_request`. Does the consumer of this endpoint know to map this message to that error?

Reply:
For invalid requests with an incorrect or blank Request Body and Client Id the only values that are returned that we can currently validate are status code and response message/data. We've validated that the response message contains  'Invalid request' but I'm not sure how we'd be able to validate that the consumer knows how to map this message to that error. @CoralSmith